### PR TITLE
debugger: implement pydevd skeleton

### PR DESCRIPTION
**Pull request by xoviat**
After some hacking around, I was able to get some basic functionality working. You should be able to merge this as it does not disable any existing functionality. In fact I would like it merged as soon as possible if it doesn't break existing functionality to avoid merge conflicts.

### CODE REVIEW

**Comment by idxn** [comment]
@xoviat You're probably busy with something else. Once you are free, please come back and get this done. I couldn't wait any longer to use the pydev debugger.

**Comment by xoviat** [comment]
> @xoviat You're probably busy with something else. Once you are free, please come back and get this done. I couldn't wait any longer to use the pydev debugger.

Fork this and find out why the tests fail; it probably has something to do with package.json. I am going to be *really* busy for the next few months, so this won't be done for a while.

**Comment by idxn** [comment]
it's ok. at least, i have some hope :)

File: src/client/extension.ts [hunk 1]
```diff
@@ -146,6 +146,17 @@ export function activate(context: vscode.ExtensionContext) {
...
 
     context.subscriptions.push(activateSingleFileDebug());
+
+    context.subscriptions.push(vscode.commands.registerCommand('extension.pydev-debug.getProgramName', config => {
+        return vscode.window.showInputBox({
+            placeHolder: "Please enter the name of a markdown file in the workspace folder",
+            value: "readme.md"
+        });
+    }));
+
```
Comment (xoviat) [inline src/client/extension.ts]: Not sure whether this is necessary? @DonJayamanne what do you think?

**Comment by mostafaeweda** [comment]
It'd be awesome to get this in as perf of the existing debugger is brutal.

**Comment by xoviat** [comment]
@DonJayamanne can you merge this so that others can work on it as well?

**Comment by xoviat** [comment]
Closing as I've received no response.

**Pull request closed without merging**